node/lib/util/push.js

```diff
@@ -65,14 +67,34 @@ exports.push = co.wrap(function *(repo, remoteName, source, target) {
 
     let errorMessage = "";
     const subRepos = yield SubmoduleUtil.getSubmoduleRepos(repo);
+    const shas = yield SubmoduleUtil.getSubmoduleShasForBranch(repo, source);
```

Comment:
Yes we do. I will put a TODO in there for now as I agree it's necessary but probably not as critical as other outstanding work.